packages/shopify-codemod/transforms/simplify-?=-if-else-statement.js

```diff
@@ -0,0 +1,62 @@
+export default function simplifyCondAssignIfElseStatement({source}, {jscodeshift: j}, {printOptions = {quote: 'single'}}) {
+
+  function testIsBinaryExpression(path) {
+    return j.BinaryExpression.check(path.get('test').node);
+  }
+
+  function testChecksIdentifierForNotNull(path) {
+    const test = path.get('test').node;
+    return (
+      j.Identifier.check(test.left) &&
```

Comment:
This should also be able to work for member expressions. I didn't check what decaf outputs, but CoffeeScript generates the following for `foo.bar.baz ?= qux`:

```js
(foo.bar.baz != null ? foo.bar.baz : foo.bar.baz = qux);
```

(as an aside, the decaf stuff all generates those ternaries — is this codemod done with the assumption that another runs before it to turn those into conditionals?)